tests/test_end_to_end.py

```diff
@@ -0,0 +1,100 @@
+"""
+End-to-end tests for each backend configuration. Sets up a server with
+the backend, sends some basic queries to that server and verifies results
+are as expected.
+"""
```

Comment:
Can we get the __future__ imports here please? These are important for py2 -> py3 compatibility, and we definitely want to support py3 at some point!